src/EventStore.Core/TransactionLog/Scavenging/InMemory/InMemoryChunkWeightScavengeMap.cs

```diff
@@ -0,0 +1,43 @@
+﻿namespace EventStore.Core.TransactionLog.Scavenging {
+	public class InMemoryChunkWeightScavengeMap :
+		InMemoryScavengeMap<int, float>,
+		IChunkWeightScavengeMap {
+
+		public bool AllWeightsAreZero() {
+			foreach (var kvp in AllRecords()) {
+				if (kvp.Value != 0) {
+					return false;
+				}
+			}
+			return true;
+		}
+
+		public void IncreaseWeight(int logicalChunkNumber, float extraWeight) {
+			// sqlite implementation would update table set weight = weight + extraWeight
```

Comment:
do the `sqlite implementation would ...` comments need to be cleaned up? it seems that they were added prior to the sqlite implementation